src/App.tsx

```diff
@@ -1,26 +1,250 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoRow } from './components/TodoRow';
+import { TodoItem } from './components/TodoItem';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { SortTodos } from './types/SortTodos';
+import { ErrorMessage } from './types/ErrorMessage';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortTodos, setSortTodos] = useState<SortTodos>('All');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processTodoIds, setProcessTodoIds] = useState<number[]>([]);
+
+  const [input, setInput] = useState('');
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [disableInput, setDisable] = useState(false);
+
+  const [error, setError] = useState<ErrorMessage>(null);
+  const [isLoading, setLoader] = useState(false);
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [changingValue, setChangingValue] = useState('');
```

Comment:
```suggestion
  const [changingValue, setChangingValue] = useState('');
```
?? let's get rid of this extra state